src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Storage.FileShare.tests.ps1

```diff
@@ -0,0 +1,98 @@
+
+                # Act
+                Create-AzFileShareStorageFolder `
+                    -ResourceGroupName $config.Arcus.ResourceGroupName `
+                    -StorageAccountName $config.Arcus.Storage.StorageAccount.Name `
+                    -FileShareName $fileShareName `
+                    -FolderName $folderName
+
+                # Assert
+                Get-AzStorageFile -ShareName $fileShareName -Context $storageAccount.Context |
+                    where { $_.GetType().Name -eq "AzureStorageFileDirectory" } |
+                    % { $_.Name } |
+                    Should -Contain $folderName
+            }
+        }
+        Context "Copy files into Azure FileShare storage folder" {
+            It "Uploads file into existing Azure FileShare storage" {
+                # Arrange
+                $folderName = "uploaded-arcus-fileshare-folder"
+                New-AzStorageDirectory -Context $storageAccount.Context -ShareName $fileShareName -Path $folderName
+
+                # Act
+                Copy-AzFileShareStorageFiles `
+                    -ResourceGroupName $config.Arcus.ResourceGroupName `
+                    -StorageAccountName $config.Arcus.Storage.StorageAccount.Name `
+                    -FileShareName $fileShareName `
+                    -SourceFolderPath "$PSScriptRoot\Blobs" `
+                    -DestinationFolderName $folderName
+
+                # Assert
+                $tempLocation = "$PSScriptRoot\arcus.png"
+                try {
+                    Get-AzStorageFileContent `
+                        -Context $storageAccount.Context `
+                        -ShareName $fileShareName `
+                        -Path "$folderName/arcus.png" `
+                        -Destination $tempLocation -Force
+                    $file = Get-Item $tempLocation
+                    $file.Length | Should -BeGreaterThan 0
+                } finally {
+                    Remove-Item $tempLocation -Force
+                }
+            }
+            It "Uploads file into non-existing Azure FileShare storage" {
+                # Arrange
+                $folderName = "non-existing-arcus-fileshare-folder"
+                $nonExistingFileShareName = "non-existing-fileshare-storage"
+
+                # Act / Assert
+                { Copy-AzFileShareStorageFiles `
+                    -ResourceGroupName $config.Arcus.ResourceGroupName `
+                    -StorageAccountName $config.Arcus.Storage.StorageAccount.Name `
+                    -FileShareName $nonExistingFileShareName `
+                    -SourceFolderPath "$PSScriptRoot\Blobs" `
+                    -DestinationFolderName $folderName } |
+                    Should -Throw
+            }
+        }
+        AfterEach {
+            Remove-AzStorageShare -Name $fileShareName -Context $storageAccount.Context -IncludeAllSnapshot -Force
```

Comment:
Removing the entire file share at each test makes sure that we don't have to remove every single file/directory ourselves.

Reply:
Is the better/quicker option indeed. 
Didn't think of that :)